perl/lib/Nix/Store.xs

```diff
@@ -17,100 +17,166 @@
 #include <sodium.h>
 #include <nlohmann/json.hpp>
 
-
-using namespace nix;
-
-
-static ref<Store> store()
-{
-    static std::shared_ptr<Store> _store;
-    if (!_store) {
-        try {
-            initLibStore();
-            _store = openStore();
-        } catch (Error & e) {
-            croak("%s", e.what());
-        }
-    }
-    return ref<Store>(_store);
-}
-
+// Everything we use but nix::Store
+using nix::CanonPath;
+using nix::CheckSigs;
+using nix::Derivation;
+using nix::DrvOutput;
+using nix::Error;
+using nix::FdSink;
+using nix::FdSource;
+using nix::FileIngestionMethod;
+using nix::FixedOutputInfo;
+using nix::Hash;
+using nix::HashFormat;
+using nix::NoCheckSigs;
+using nix::PosixSourceAccessor;
+using nix::SecretKey;
+using nix::StorePathSet;
+using nix::Verbosity;
+using nix::baseNameOf;
+using nix::initLibStore;
+using nix::openStore;
+using nix::parseHashAlgo;
+using nix::ref;
+using nix::settings;
+using nix::verbosity;
```

Comment:
Is this enumeration really needed? Why not just `using namespace nix`?

Reply:
@edolstra I would rather do that, but then the `Store` wrapper `struct conflicts. I am not sure whether the name of that struct matters, however.